src/Mod/Sketcher/Gui/SketcherSettings.cpp

```diff
@@ -126,12 +127,80 @@ SketcherSettingsDisplay::SketcherSettingsDisplay(QWidget* parent)
 
         QPainter painter(&px);
         painter.setPen(pen);
-        double mid = ui->comboBox->iconSize().height() / 2.0;
-        painter.drawLine(0, mid, ui->comboBox->iconSize().width(), mid);
+        double mid = ui->linePatternComboBox->iconSize().height() / 2.0;
+        painter.drawLine(0, mid, ui->linePatternComboBox->iconSize().width(), mid);
         painter.end();
 
-        ui->comboBox->addItem(QIcon(px), QString(), QVariant(it->second));
+        ui->linePatternComboBox->addItem(QIcon(px), QString(), QVariant(it->second));
+        ui->lineDivisionPatternComboBox->addItem(QIcon(px), QString(), QVariant(it->second));
     }
+}
+
+SketcherSettingsGrid::~SketcherSettingsGrid()
+{
+    // no need to delete child widgets, Qt does it all for us
+}
+
+void SketcherSettingsGrid::saveSettings()
+{
+    ui->gridSmallLineWidth->onSave();
+    ui->gridBigLineWidth->onSave();
+    ui->gridNumberSubdivision->onSave();
+    ui->checkBoxShowGrid->onSave();
+    ui->gridSize->onSave();
+    ui->checkBoxGridSnap->onSave();
+    ui->checkBoxGridAuto->onSave();
+
+    ParameterGrp::handle hGrp = App::GetApplication().GetParameterGroupByPath("User parameter:BaseApp/Preferences/Mod/Part");
```

Comment:
Hmmm. I really don't like that Sketcher settings are modifying general settings that can have an impact on other objects.

Reply:
I understand. 
But I don't understand: Is the grid actually used by something else than sketcher? 
Can't we just have a sketcher grid. It's not even shared with draft which is the only other workbench where I could think of a grid.

Reply:
> But I don't understand: Is the grid actually used by something else than sketcher?

I'd like that have checked by yourself.
Currently, it doesn't seem to be used by other objects in master codebase. Then, who knows for addon workbenches ???

Reply:
In my opinion having the grid as such a 'shared object' in 'Part' is very counterproductive if this means that we can't modify it anymore on the hypothesis that something else external to freecad, maybe uses it.

Reply:
Sure I can add a setting to toggle the division lines.

Reply:
Sorry I was unclear. The subdivision marking shall be disabled by default in `ViewProvider2DObject` (only).
Then in Sketcher you can enable it by default, not a problem for me.
Another advantage is that in this case, this is a Sketcher specific setting, which is good. ;)